src/vanilla/utils/atomWithStorage.ts

```diff
@@ -13,6 +13,7 @@ export interface AsyncStorage<Value> {
   getItem: (key: string, initialValue: Value) => Promise<Value>
   setItem: (key: string, newValue: Value) => Promise<void>
   removeItem: (key: string) => Promise<void>
+  getItemOnInit?: boolean
```

Comment:
I'm open for changing this property name.
It would be helpful if `createJSONStorage` takes it as an option.